src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Circle circle = new Circle(colorSupplier.getRandomColor(), new Random().nextInt(100));
+        IsoscelesTrapezoid isoscelesTrapezoid =
+                new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                new Random().nextInt(100),
+                new Random().nextInt(100),
+                new Random().nextInt(100));
+        Rectangle rectangle = new Rectangle(colorSupplier.getRandomColor(),
+                new Random().nextInt(100),
+                new Random().nextInt(100));
+        RightTriangle rightTriangle = new RightTriangle(colorSupplier.getRandomColor(),
+                new Random().nextInt(100),
+                new Random().nextInt(100));
+        Square square = new Square(colorSupplier.getRandomColor(),
+                new Random().nextInt(100));
+        Figure[] figures = {circle, isoscelesTrapezoid, rectangle, rightTriangle, square};
```

Comment:
We create here 5 shapes in order to choose one of them (it turns out we don’t use the other 4 anywhere). Let's create 5 private methods in this class getRandomCircle(), getRandomSquare(),... and call these methods in the switch-case construction.